### DISCUSSION TITLE
Netbox in docker with nginx

### DISCUSSION THREAD

**Discussion by jonare77** [netbox-community/netbox#9978] [General]
Hello

I am running netbox in docker and trying to get nginx (container: jwilder/nginx-proxy) to work as proxy.
I have used the overide file to specify the needed parameters as usuall. But get "connection refused" in the nginx container. 
The IP and ports are correct and the nginx is trying the right address. The configuration.py is also containing the correct ALLOW=* option. 

Not understanding what I am missing. Anyone have any suggestions ?

Thanks.

**Comment by candlerb** [comment]
There is a working docker setup here: https://github.com/netbox-community/netbox-docker

This runs nginx *within* the main netbox docker container (so that it can serve static assets).

You can put another reverse proxy in front of this if you like - it sounds like you're running another docker container for this. But how you configure that front-end proxy is up to you, and you'll need to debug it yourself.  If you're getting "connection refused" this means that either the back-end isn't running, or you're pointing the front-end to the wrong back-end URL or port, or there's some sort of networking issue between them (e.g. firewalling).

Unless netbox itself isn't running, then this isn't an issue with netbox itself: it's an issue with your proxy, which you'll need to solve locally. You may find "tcpdump" helpful to inspect the traffic being sent from proxy to netbox.  Good luck!

**Reply by jonare77** [reply]
I am running the stack you are pointing to, the backend is running at port 8000 and working. The containers are also in the same subnet with correct IP's. No firewall or network problems between the containers. Thats why I am asking, but will keep looking. Thank you.

**Reply by jackyscript** [reply]
Possibly to late at this issue, but I hope this helps someone as it helped me, after I searched for a solution.

I had a similar problem and resolved it by connecting my nginx container with the netbox container.
Check the available networks by entering `docker network ls` and make sure the nginx container is connected to the same network as the netbox container, using `docker inspect`. Prior to this, this was not the case so there was no chance to establish a connection.

**Comment by lehnbe** [comment]
I have an additional question. Where is the nginx.conf file? I followed up the instruction on
 https://github.com/netbox-community/netbox-docker
I now want to setup nginx that I can access netbox on a different Port.  Maybe on Port 443 with https. I didn't find an article about the "docker included"nginx configuration
